dotCMS/src/main/java/com/dotcms/graphql/datafetcher/NavigationDataFetcher.java

```diff
@@ -0,0 +1,107 @@
+package com.dotcms.graphql.datafetcher;
+
+import static com.dotcms.rest.api.v1.page.NavResource.navToMap;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotcms.graphql.exception.InvalidLanguageException;
+import com.dotcms.graphql.exception.ResourceNotFoundException;
+import com.dotcms.rendering.velocity.viewtools.navigation.NavResult;
+import com.dotcms.rendering.velocity.viewtools.navigation.NavTool;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.VelocityUtil;
+import com.liferay.util.StringPool;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.velocity.tools.view.context.ChainedContext;
+import org.apache.velocity.tools.view.context.ViewContext;
+
+/**
+ * DataFetcher for the Navigation GraphQL type.
+ */
+public class NavigationDataFetcher implements DataFetcher<Map<String, Object>>  {
+
+    /**
+     * This method is called by the GraphQL engine to fetch the data for the field.
+     * @param environment this is the data fetching environment which contains all the context you need to fetch a value
+     *
+     * @return a map containing the navigation data
+     * @throws Exception if an error occurs while fetching the data
+     */
+    @Override
+    public Map<String, Object> get(DataFetchingEnvironment environment) throws Exception {
+        final long defLangId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        final int maxDepth = 1;
+        final DotGraphQLContext context = environment.getContext();
+        final HttpServletRequest request = context.getHttpServletRequest();
+        final HttpServletResponse response = context.getHttpServletResponse();
+        final long languageId = getArgument(environment, "languageId", defLangId);
+        final String uri = getArgument(environment, "uri", StringPool.FORWARD_SLASH);
+        final Long depth = getArgument(environment, "depth", (long) maxDepth);
```

Comment:
would be great to ask us if we want to allow more than 1 depth here, since it is pretty expensive